test/modules/obsagent-client/src/config.rs

```diff
@@ -43,10 +45,12 @@ impl Config {
             return Err(ConfigError::InvalidNegativeArgVal(update_period));
         } else if update_period == 0.0 {
             return Err(ConfigError::InvalidZeroArgVal(update_period));
+        } else if update_period == f64::INFINITY {
+            return Err(ConfigError::InvalidInfiniteArgVal(update_period));
         }
 
         Ok(Config {
-            update_period,
```

Comment:
nit: in the constructor, this can be taken in as a `Duration` (and parsed from the env vars) rather than an `f64`